partner_multi_relation_parent/models/res_partner_relation.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import _, api, models
+from odoo.exceptions import UserError
+
+
+class ResPartnerRelation(models.Model):
+    _inherit = 'res.partner.relation'
+
+    def relation_exists(self, left, type_rel, right):
+        relation = self.search([
+            ('left_partner_id', '=', left),
+            ('type_id', '=', type_rel),
+            ('right_partner_id', '=', right)
+        ])
+        return relation
+
+    @api.multi
+    def write(self, vals):
+        part_mod = self.env['res.partner']
+        type_relation = self.env.ref(
+            'partner_multi_relation_parent.parent_relation_type'
+        ).id
+        for this in self:
+            if this.type_id != type_relation:
+                continue
+            # check that whatever relation will come out of this write does
+            # not exist already , but check only if type_id doesn't change, if
+            # it does we don't care about uniqueness
+            if vals.get('type_id', this.type_id) == type_relation:
+                relation = this.relation_exists(
+                    vals.get('left_partner_id', this.left_partner_id),
+                    type_relation,
+                    vals.get('right_partner_id', this.right_partner_id)
+                )
+                if relation:
```

Comment:
this is logaclly wrong. You need to check that the left partner has no other relations of this type, not only no relations with the current partner

Reply:
checking all partners